examples/DST_Europe/DST_Europe.ino

```diff
@@ -0,0 +1,96 @@
+/* European Daylight Saving Time (DST) example */
+
+// The Real Time Clock is kept in local winter time (standard time). This approach is key for the simplicity of this solution.
+// This code is for a DST from Last sunday in March 1:00 (am) local time until Last sunday in October 3:00 (am) local time. 
+// The DST time can be changed if required, by choosing a different hour in the DateTime function (4th parameter).
+
+#include <RTClib.h>   // https://github.com/adafruit/rtclib 
+RTC_DS3231 rtc; 
+//RTC_DS1307 rtc;
+
+#define USEDST  true // Use DST (true or false)
+#define BAUD   57600 // Baud rate for the serial monitor
+
+DateTime now; 
+int lastsec;
+
+DateTime dstclock(DateTime n) { // Return the given (DST adjusted) date and time according to DST settings (for extensive date and time calculations)
+  
+  DateTime b, e;
+  
+  b = DateTime(n.year(), 3, 31, 1, 0, 0); // Begin of DST: set on March 31 1:000 (am)
+  if (n.month() == 3) b = DateTime(n.year(), 3, 31 - b.dayOfTheWeek(), 1, 0, 0); // Begin of DST: adjusted to last sunday in March 1:00 (am) when actual month is March
+  e = DateTime(n.year(), 10, 31, 3, 0, 0); // End of DST: set on October 31 3:00 (am)
+  if (n.month() == 10) e = DateTime(n.year(), 10, 31 - e.dayOfTheWeek(), 3, 0, 0); // End of DST: adjusted to last sunday in October 3:00 (am) when actual month is October
+
+  if (USEDST && (n > b) && (n < e)) n = n + TimeSpan(0,1,0,0); // adjust to standard time if within summertime
+  return n;
+}; 
+
+DateTime getclock() { // Retrieve the (DST adjusted) date and time
+  
+  DateTime n, b, e;
+  
+  n = rtc.now();
+  b = DateTime(n.year(), 3, 31, 1, 0, 0); // Begin of DST: set on March 31 1:000 (am)
+  if (n.month() == 3) b = DateTime(n.year(), 3, 31 - b.dayOfTheWeek(), 1, 0, 0); // Begin of DST: adjusted to last sunday in March 1:00 (am) when actual month is March
+  e = DateTime(n.year(), 10, 31, 3, 0, 0); // End of DST: set on October 31 3:00 (am)
+  if (n.month() == 10) e = DateTime(n.year(), 10, 31 - e.dayOfTheWeek(), 3, 0, 0); // End of DST: adjusted to last sunday in October 3:00 (am) when actual month is October
+
+  if (USEDST && (n > b) && (n < e)) n = n + TimeSpan(0,1,0,0); // adjust to standard time if within summertime
+  return n;
+}; 
```

Comment:
Replace this function with

```c++
DateTime getclock() { // Retrieve the (DST adjusted) date and time
  return dstclock(rtc.now());
};
```

Reply:
I realized that this function could be replaced with:

#define getclock() dstclock(rtc.now()) 

Do you want me to make that change?

Reply:
I have a slight preference for a function over a #define, as a function is more type-safe and more C++-idiomatic. But I don't care that much, and I am fine with whatever option you prefer.